gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ProjectInfo.java

```diff
@@ -0,0 +1,355 @@
+   */
+  public String projectId() {
+    return projectId;
+  }
+
+  /**
+   * Get the user-assigned name of the project.
+   *
+   * This field is optional, can remain unset, and can be changed after project creation.
+   */
+  public String name() {
+    return Data.isNull(name) ? null : name;
+  }
+
+  /**
+   * Get number uniquely identifying the project.
+   *
+   * This field is set by the server and is read-only.
+   */
+  public Long projectNumber() {
+    return projectNumber;
+  }
+
+  /**
+   * Get the immutable map of labels associated with this project.
+   */
+  public Map<String, String> labels() {
+    return labels;
+  }
+
+  /**
+   * Get the project's lifecycle state.
+   *
+   * This is a read-only field. To change the lifecycle state of your project, use the
+   * {@code delete} or {@code undelete} method.
+   */
+  public State state() {
+    return state;
+  }
+
+  ResourceId parent() {
+    return parent;
+  }
+
+  /**
+   * Get the project's creation time (in milliseconds).
+   *
+   * This field is set by the server and is read-only.
+   */
+  public Long createTimeMillis() {
+    return createTimeMillis;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj instanceof ProjectInfo && Objects.equals(toPb(), ((ProjectInfo) obj).toPb());
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(name, projectId, labels, projectNumber, state, createTimeMillis, parent);
+  }
+
+  public static Builder builder(String id) {
+    return new Builder().projectId(id);
+  }
+
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  com.google.api.services.cloudresourcemanager.model.Project toPb() {
+    com.google.api.services.cloudresourcemanager.model.Project projectPb =
+        new com.google.api.services.cloudresourcemanager.model.Project();
+    projectPb.setName(name);
+    projectPb.setProjectId(projectId);
+    projectPb.setLabels(labels);
+    projectPb.setProjectNumber(projectNumber);
+    if (state != null) {
+      projectPb.setLifecycleState(state.toString());
+    }
+    if (createTimeMillis != null) {
+      projectPb.setCreateTime(ISODateTimeFormat.dateTime().print(createTimeMillis));
+    }
+    if (parent != null) {
+      projectPb.setParent(parent.toPb());
+    }
+    return projectPb;
+  }
+
+  static ProjectInfo fromPb(com.google.api.services.cloudresourcemanager.model.Project projectPb) {
+    ProjectInfo.Builder builder =
+        ProjectInfo.builder(projectPb.getProjectId()).projectNumber(projectPb.getProjectNumber());
+    if (projectPb.getName() != null) {
+      builder.name(projectPb.getName());
+    }
+    if (projectPb.getLabels() != null) {
+      builder.labels(projectPb.getLabels());
+    }
+    if (projectPb.getLifecycleState() != null) {
+      builder.state(State.valueOf(projectPb.getLifecycleState()));
+    }
+    if (projectPb.getCreateTime() != null) {
+      builder.createTimeMillis(DateTime.parse(projectPb.getCreateTime()).getMillis());
```

Comment:
Great, in this API we'll represent the time as milliseconds since epoch then.  Are there plans to support microseconds or nanoseconds in the future?  I ask because the docs linked above mention it.